### DISCUSSION TITLE
(GoogleChrome) Windows Server 2012 R2 and Windows 8.1 Chrome security patch builds

### DISCUSSION THREAD

**Discussion by AdamDeCelis** [chocolatey-community/chocolatey-packages#2248] [Ideas]
### Before starting, ensure that you have done the following.

- [X] I have verified that there is no other issue or discussion covering the feature or enhancement
- [X] I have verified there is no open pull request for the feature or enhancement
- [X] I have verified the package to improve is located in this repository

### What do you want to add?

I'd like to request that version 109.0.5414.149 be made available as updates for systems running Windows Server 2012R2 and Windows 8.1 as these include security patches for the recent Chrome exploits and these and older Windows OS versions are unable to update to Google Chrome 110+

The latest version that these OS's can update to via Chocolatey is 109.0.5414.120 which leaves systems vulnerable

Google provide these special releases for older Windows operating systems from the below URLs: 

[Google Chrome 32 Bit MSI - Build 109](https://edgedl.me.gvt1.com/edgedl/release2/10/windows-8/googlechromestandaloneenterprise.msi)
[Google Chrome 64 Bit MSI - Build 109](https://edgedl.me.gvt1.com/edgedl/release2/10/windows-8/googlechromestandaloneenterprise64.msi)

See article from Google here:

[Sunsetting support for Windows 7 / 8/8.1 and Windows Server 2012 and Windows Server 2012 R2 in 2023](https://support.google.com/chrome/a/thread/185534987)

Google are providing critical security patch versions for these OS versions until they are EOL in October 2023.

### What problem will the feature/enhancement solve?

This will allow for chocolatey to patch Google Chrome versions on older operating systems to include critical security fixes for known exploits 

### If the feature or enhancement is approved:

- [X] I am willing to create a pull request to implement the feature or enhancement

### Additional Context

This also affects other Chromium based browsers such as Edge and Brave. But Google Chrome is the most common so I thought it best to have it implemented here first.

Feel free to contact me here, or via the Disqus chat in the choco repository if you need further information.

**Comment by pauby** [comment]
As Google only produces one setup file, it's not possible to have specific versions in the packages. As that version had already been approved, it's not possible to go and update it with the installer versions you mentioned.

It may be possible to push a package fix version that would be 109.0.5414.121 (note the last number is incremented).

That would be a very manual process that one of the repository maintainers would have to agree to pick up as I don't think we could have a PR for it as that would affect the automation.

**Reply by TheCakeIsNaOH** [reply]
The special build appears to be `109.0.5414.149`, not a second rev of `109.0.5414.120`, so no need for a package fix version.

**Reply by AdamDeCelis** [reply]
I know it's not long until Google and Microsoft stop providing support for Win 8.1 and server 2012 R2 in October.

I just don't like the idea that people are assuming they're getting the latest version available for those OS's but could be vulnerable and unaware of it. Especially if they didn't do a choco upgrade until after the main version exceeded the version they could install.

They could be stuck on any previous version and choco won't upgrade them to 109 because 110+ is available but fails to install for them. Let alone the fact that 109 isn't the latest available 109 released. 
I know chrome auto updates, but people may have disabled that and are relying on choco.

**Reply by pauby** [reply]
There are a couple of options here:

1. Support is built into the package to not update those operating systems beyond the supported package version. We would also need to decide if the package upgrade fails or success at that point. I can see arguments for both.
2. End users pin their package version at the final supported version so they don't upgrade beyond it.

I prefer 2. I think that's the sensible, consistent and quickest solution.

**Reply by AdamDeCelis** [reply]
They can't update past 109 versions it fails ever since 110 was released, they can't get to the latest version of 109 even if they are on 108 unless they specify the version number

**Reply by AdamDeCelis** [reply]
And the latest 109 available in choco isn't the latest version Google host

**Reply by pauby** [reply]
> They can't update past 109 versions it fails ever since 110 was released, they can't get to the latest version of 109 even if they are on 108 unless they specify the version number

That's why I suggested they pin at that package version.

> And the latest 109 available in choco isn't the latest version Google host

That's why I suggested that the version be pushed above and @TheCakeIsNaOH highlighted that the version the Chocolatey Community Repository have is the latest.

**Reply by pauby** [reply]
I've created a branch over on [my packages' repository for this](https://github.com/pauby/ChocoPackages/tree/temp-coreteam-chrome-8.1-update/dev/googlechrome). There may be a better way of doing it, but I didn't want to disturb anything in here.

The package has been tested on the Chocolatey Test Environment (Windows Server 2019). I don't have a VM that this version is specifically for, but it shouldn't make any difference. It installs version `109.0.5414.149`. However, it would be good if others could test this.

I am happy to be added as a maintainer, push this version, and then remove myself again.

**Reply by pauby** [reply]
@AdamDeCelis are you still interested in this package?

**Reply by AdamDeCelis** [reply]
Cheers for that... I'll check it out next time I have to work on a 2012 server
